### ISSUE TITLE
Moving already enrolled account to a new OU causes enrollement to fail

### ISSUE THREAD

**Issue by CalvinRodo** [aws-ia/terraform-aws-control_tower_account_factory#286]
**Terraform Version & Prov:**

**AFT Version:**
1.6.2

**Terraform Version & Provider Versions**
Please provide the outputs of `terraform version` and `terraform providers` from within your AFT environment

`terraform version`
```
1.1.7
```

`terraform providers`
```
{Replace me}
```

**Bug Description**
When moving an already enrolled account from one OU to another the account fails to get enrolled in the new account. 

We receive the following error in Control Tower: 

> AWS Control Tower could not enroll your account for the following reason: AWS Control Tower cannot create the IAM role aws-controltower-ReadOnlyExecutionRole because the role already exists. To continue, delete the existing IAM role and try again.

**To Reproduce**
1. Change OU in AFT Module for account
2. Merge changes
3. Check control tower for enrollment issues

**Expected behavior**
No issues identified when enrolling

**Related Logs**
Provide any related logs or error messages to help explain your problem.

**Additional context**

**Comment by CalvinRodo** [comment]
So moving OUs using AFT left us in a situation where we had to manually delete all of the control tower resources, this was the outcome of several hours. chatting with our premium support repo. 

Your [documentation](https://docs.aws.amazon.com/controltower/latest/userguide/aft-update-account.html) states that we can change this value but does not state that this will result in your account's enrollment being broken and require manual deletion of all Control Tower managed resources. 

>... AFT allows you to update the ManagedOrganizationalUnit field of the control_tower_parameters input...

Would it be possible to update your documentation so that either it provides instructions on how to do this without breaking things (Terminate account in service catalogue, and then manually move the account before re-enrolling in the new account) or to state that it will result in your account being in a broken state when trying to re-enroll.

**Comment by v-rosa** [comment]
Just to add my personal experience, I've tested this flow of updating OUs and it worked perfectly, yet I was using v1.4.2.

**Comment by CalvinRodo** [comment]
Yeah I could have sworn I had successfully done it before in the past, I'll give it a try with a new account to see what happens.

**Comment by stumins** [comment]
We move accounts between OUs as part of regular testing and have been unable to replicate this issue internally.

@CalvinRodo I'm going to close this issue but please feel free to open a new issue with steps for us reproduce if you're continuing to experience this issue.
